parustiko/src/protocol/binary_packet.rs

```diff
@@ -119,6 +123,18 @@ impl DecodeRaw for BinaryProtocolPacket {
     }
 }
 
+impl Decode for BinaryProtocolPacket {
+    type Entity = Self;
+
+    fn from_be_bytes(buffer: Vec<u8>) -> Result<Self::Entity, BppError> {
+        let mut cursor = std::io::Cursor::new(buffer);
```

Comment:
why cursor? cant we use a simple slice?

Reply:
actually we can, I thought the cursor as a pointer would be lighter, but as I see, the slice is even better